### PR TITLE
Add integration tests for group user operations

### LINKED ISSUE #93: Group integration test: Group User Operations
- [ ]  Create a user with credentials → POST /api/v1/users?activate=false

```js
const newUser = {
    profile: {
        firstName: 'John',
        lastName: 'With-Group',
        email: 'john-with-group@example.com',
        login: 'john-with-group@example.com'
    },
    credentials: {
        password: { value: 'Abcd1234' }
    }
};
```

- [x]  Create a new group → POST /api/v1/groups

```js
const newGroup = {
    profile: {
        name: 'Group-Member API Test Group'
    }
};
```

- [x] Add user to the group and validate user present in group → POST /api/v1/groups/{{groupId}}/users/{{userId}} + GET /api/v1/groups/{{groupId}}/users

- [x]  Remove user from group and validate user removed → DELETE  /api/v1/groups/{{groupId}}/users/{{userId}} + GET /api/v1/groups/{{groupId}}/users

- [x]  Delete the group and user → POST /api/v1/users/{{userId}}/lifecycle/deactivate + DELETE /api/v1/users/{{userId}} +  DELETE /api/v1/groups/{{groupId}}

### PR DESCRIPTION
**Pull request by laura-rodriguez**
Resolves
#93

### CODE REVIEW

File: src/Okta.Sdk.IntegrationTests/GroupScenarios.cs [hunk 1]
```diff
@@ -119,10 +119,224 @@ public class GroupScenarios : ScenarioGroup
...
+            });
+
+            // Create a user
+            var createdUser = await client.Users.CreateUserAsync(new CreateUserWithPasswordOptions
+            {
+                Profile = new UserProfile
+                {
+                    FirstName = "John",
+                    LastName = "Get-User",
+                    Email = "john-get-user@example.com",
```
Comment (nbarbettini) [inline src/Okta.Sdk.IntegrationTests/GroupScenarios.cs]: Rename email and login to `john-add-group@example.com` (these different emails are intended to reduce collisions if the IT doesn't clean up after itself properly for some reason)

File: src/Okta.Sdk.IntegrationTests/GroupScenarios.cs [hunk 2]
```diff
@@ -119,10 +119,224 @@ public class GroupScenarios : ScenarioGroup
...
```
Comment (nbarbettini) [inline src/Okta.Sdk.IntegrationTests/GroupScenarios.cs]: Rename this email and login as well

File: src/Okta.Sdk.IntegrationTests/GroupScenarios.cs [hunk 3]
```diff
@@ -119,10 +119,224 @@ public class GroupScenarios : ScenarioGroup
...
+                    Login = "john-get-user@example.com",
+                },
+                Password = "Abcd1234",
+                Activate = true,
+            });
+
+            await client.Groups.AddUserToGroupAsync(createdGroup.Id, createdUser.Id);
```
Comment (nbarbettini) [inline src/Okta.Sdk.IntegrationTests/GroupScenarios.cs]: I think these lines should also be within the `try`. That way if it fails at the beginning, things are still cleaned up. What do you think? (applies to all these new tests)

**Comment by nbarbettini** [comment]
Looking good, a few comments. Can you split this up from the other code changes?

File: src/Okta.Sdk.IntegrationTests/GroupScenarios.cs (hunk 3 above)
Comment (laura-rodriguez) [reply src/Okta.Sdk.IntegrationTests/GroupScenarios.cs]: if I want `createdUser` to be available in the `finally` scope, it should be declared before `try` block
@nbarbettini

Comment (nbarbettini) [reply src/Okta.Sdk.IntegrationTests/GroupScenarios.cs]: I should have been more specific - I meant just the `AddUserToGroup` line. You understood what I meant 🙂  👍

**Comment by nbarbettini** [comment]
Superseded by #237

**Pull request closed without merging**